go/src/github.com/mesosphere/rendler/scheduler/main.go

```diff
@@ -1,281 +1,284 @@
 package main
 
 import (
-	"code.google.com/p/goprotobuf/proto"
 	"container/list"
 	"encoding/json"
 	"flag"
 	"fmt"
-	"github.com/mesosphere/mesos-go/mesos"
+	"github.com/golang/protobuf/proto"
+	mesos "github.com/mesos/mesos-go/mesosproto"
+	"github.com/mesos/mesos-go/mesosutil"
+	sched "github.com/mesos/mesos-go/scheduler"
 	"github.com/mesosphere/rendler"
 	"log"
 	"os"
 	"os/signal"
-	"time"
 	"path/filepath"
+	"time"
 )
 
-const TASK_CPUS = 0.1
-const TASK_MEM = 32.0
-const SHUTDOWN_TIMEOUT = 30  // in seconds
+const (
+	TaskCPUs        = 0.1
+	TaskMem         = 32.0
+	ShutdownTimeout = time.Duration(30) * time.Second
+)
 
-func main() {
-	crawlQueue := list.New()  // list of string
-	renderQueue := list.New() // list of string
+const (
+	crawlCommand  = "python crawl_executor.py"
+	renderCommand = "python render_executor.py"
+)
 
-	processedURLs := list.New() // list of string
-	crawlResults := list.New()  // list of CrawlEdge
-	renderResults := make(map[string]string)
+var (
+	DefaultFilter = &mesos.Filters{RefuseSeconds: proto.Float64(1)}
+)
 
-	seedUrl := flag.String("seed", "http://mesosphere.io", "The first URL to crawl")
-	master := flag.String("master", "127.0.1.1:5050", "Location of leading Mesos master")
-	localMode := flag.Bool("local", true, "If true, saves rendered web pages on local disk")
-	// TODO(nnielsen): Add flag for artifacts.
+// maxTasksForOffer computes how many tasks can be launched using a given offer
+func maxTasksForOffer(offer *mesos.Offer) int {
+	count := 0
 
-	flag.Parse()
+	var cpus, mem float64
 
-	crawlQueue.PushBack(*seedUrl)
+	for _, resource := range offer.Resources {
+		switch resource.GetName() {
+		case "cpus":
+			cpus = *resource.GetScalar().Value
+		case "mem":
+			mem = *resource.GetScalar().Value
+		}
+	}
 
-	tasksCreated := 0
-	tasksRunning := 0
-	shuttingDown := false
+	for cpus >= TaskCPUs && mem >= TaskMem {
+		count++
+		cpus -= TaskCPUs
+		mem -= TaskMem
+	}
 
-	crawlCommand := "python crawl_executor.py"
-	renderCommand := "python render_executor.py"
+	return count
+}
 
-	if *localMode {
-		renderCommand += " --local"
-	}
+// RendlerScheduler implements the Scheduler interface and stores
+// the state needed for Rendler to function.
```

Comment:
Document what this is first, what it does second, and if really needed, how it does it.